client/java/build.gradle

```diff
@@ -127,6 +127,28 @@ task sourceJar(type: Jar) {
     from sourceSets.main.allJava
 }
 
+javadoc {
+    dependsOn delombok
+    failOnError = true
+
+    def capturedOutput = []
+    def listener = { capturedOutput << it } as StandardOutputListener
+    doFirst {
+        logging.addStandardErrorListener(listener)
+        logging.addStandardOutputListener(listener)
+    }
+    doLast {
+        logging.removeStandardOutputListener(listener)
+        logging.removeStandardErrorListener(listener)
+        capturedOutput.each { e ->
+            if(e.toString() =~ " warning: " && !(e.toString() =~ "Config.java")) {
+                // doesn't count *Config.java classes with use failing `@with` annotation
```

Comment:
This still leaves 12 warnings being printed...

Possible solutions are: 
 * exclude javadoc for those config classes
 * get rid of Lombok `with` which is used in few places in Spark & Flink integration

Leaving this as is for now. Perhaps new Lombok version will write better docs.